RecoTracker/LSTCore/src/alpaka/MiniDoublet.h

```diff
@@ -0,0 +1,914 @@
+                                                    float shiftedY,
+                                                    float shiftedZ,
+                                                    float noShiftedDphi,
+                                                    float noShiftedDPhiChange,
+                                                    unsigned int idx) {
+    //the index into which this MD needs to be written will be computed in the kernel
+    //nMDs variable will be incremented in the kernel, no need to worry about that here
+
+    mds.moduleIndices()[idx] = lowerModuleIdx;
+    unsigned int anchorHitIndex, outerHitIndex;
+    if (modules.moduleType()[lowerModuleIdx] == PS and modules.moduleLayerType()[lowerModuleIdx] == Strip) {
+      mds.anchorHitIndices()[idx] = upperHitIdx;
+      mds.outerHitIndices()[idx] = lowerHitIdx;
+
+      anchorHitIndex = upperHitIdx;
+      outerHitIndex = lowerHitIdx;
+    } else {
+      mds.anchorHitIndices()[idx] = lowerHitIdx;
+      mds.outerHitIndices()[idx] = upperHitIdx;
+
+      anchorHitIndex = lowerHitIdx;
+      outerHitIndex = upperHitIdx;
+    }
+
+    mds.dphichanges()[idx] = dPhiChange;
+
+    mds.dphis()[idx] = dPhi;
+    mds.dzs()[idx] = dz;
+    mds.shiftedXs()[idx] = shiftedX;
+    mds.shiftedYs()[idx] = shiftedY;
+    mds.shiftedZs()[idx] = shiftedZ;
+
+    mds.noShiftedDphis()[idx] = noShiftedDphi;
+    mds.noShiftedDphiChanges()[idx] = noShiftedDPhiChange;
+
+    mds.anchorX()[idx] = hits.xs()[anchorHitIndex];
+    mds.anchorY()[idx] = hits.ys()[anchorHitIndex];
+    mds.anchorZ()[idx] = hits.zs()[anchorHitIndex];
+    mds.anchorRt()[idx] = hits.rts()[anchorHitIndex];
+    mds.anchorPhi()[idx] = hits.phis()[anchorHitIndex];
+    mds.anchorEta()[idx] = hits.etas()[anchorHitIndex];
+    mds.anchorHighEdgeX()[idx] = hits.highEdgeXs()[anchorHitIndex];
+    mds.anchorHighEdgeY()[idx] = hits.highEdgeYs()[anchorHitIndex];
+    mds.anchorLowEdgeX()[idx] = hits.lowEdgeXs()[anchorHitIndex];
+    mds.anchorLowEdgeY()[idx] = hits.lowEdgeYs()[anchorHitIndex];
+    mds.anchorHighEdgePhi()[idx] = alpaka::math::atan2(acc, mds.anchorHighEdgeY()[idx], mds.anchorHighEdgeX()[idx]);
+    mds.anchorLowEdgePhi()[idx] = alpaka::math::atan2(acc, mds.anchorLowEdgeY()[idx], mds.anchorLowEdgeX()[idx]);
+
+    mds.outerX()[idx] = hits.xs()[outerHitIndex];
+    mds.outerY()[idx] = hits.ys()[outerHitIndex];
+    mds.outerZ()[idx] = hits.zs()[outerHitIndex];
+    mds.outerRt()[idx] = hits.rts()[outerHitIndex];
+    mds.outerPhi()[idx] = hits.phis()[outerHitIndex];
+    mds.outerEta()[idx] = hits.etas()[outerHitIndex];
+    mds.outerHighEdgeX()[idx] = hits.highEdgeXs()[outerHitIndex];
+    mds.outerHighEdgeY()[idx] = hits.highEdgeYs()[outerHitIndex];
+    mds.outerLowEdgeX()[idx] = hits.lowEdgeXs()[outerHitIndex];
+    mds.outerLowEdgeY()[idx] = hits.lowEdgeYs()[outerHitIndex];
+  }
+
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE float isTighterTiltedModules(ModulesConst modules, uint16_t moduleIndex) {
```

Comment:
Seems like the return value type should be `bool`
```suggestion
  ALPAKA_FN_ACC ALPAKA_FN_INLINE bool isTighterTiltedModules(ModulesConst modules, uint16_t moduleIndex) {
```

Reply:
(this `bool is*` was here in the original PR submission in June; and in the base code since conception)
can the static check or something similar pick up cases like this?
I've seen other cases in different code; sometimes it takes an effort to notice.

Reply:
> can the static check or something similar pick up cases like this?

I see clang-tidy has a checker for implicit `bool` conversions https://clang.llvm.org/extra/clang-tidy/checks/readability/implicit-bool-conversion.html. We could look into that.